pallets/messages/src/lib.rs

```diff
@@ -362,7 +367,17 @@ impl<T: Config> Pallet<T> {
 				list.len().try_into().map_err(|_| Error::<T>::TypeConversionOverflow)?;
 			for i in from_index..list_size {
 				let m = list[i as usize].clone();
-				response.content.push(m.map_to_response(block_number));
+
+				let mut payload_length: u32 = 0;
+				let payload = m.payload.clone().into_inner();
+
+				if payload_location == PayloadLocation::OnChain {
+					payload_length = m.payload.len().try_into().unwrap();
+				} else if payload_location == PayloadLocation::IPFS {
+					type PayloadType = (Vec<u8>, u32);
+					(_, payload_length) = PayloadType::decode(&mut &payload[..]).unwrap();
```

Comment:
Where is the `cid` coming from in the response if not here?

Reply:
This was written under the assumption that the client would decode the CID from the payload, but I think it makes sense to get the cid here and assign it to the payload in `MessageResponse`.